### PR TITLE
Make No Empty Line Required Before GFM Tables

### LINKED ISSUE #1733: Gfm Tables Incorrectly Require Newlines Before Table
gfm tables using marked require a line with just whitespace before the table definition:

```
test
|a|b|
|-|-|
```

renders just fine in GitHub:

test
|a|b|
|-|-|

However, marked doesn't like it and doesn't render a table:
[Marked Demo](https://marked.js.org/demo/?text=test%0A%7Ca%7Cb%7C%0A%7C-%7C-%7C&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)

```
test

|a|b|
|-|-|
```

works just fine both in marked and in GitHub:

test

|a|b|
|-|-|

(above renders table as expected)

**Expected behavior**
marked's lexer is able to identify tables, even when the preceding line contains anything other than whitespace.

### PR DESCRIPTION
**Pull request by EmmetZC**
Uses `table` pattern to interrupt `paragraph`. So an empty line won't be required for GFM tables.
Test cases added for this change.

**Marked version:** 4.0.4

**Markdown flavor:** GitHub Flavored Markdown

## Description

- Fixes #1733 

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

File: test/specs/gfm/gfm.0.29.json [hunk 1]
```diff
@@ -47,6 +47,18 @@
     "markdown": "| abc | def |\n| --- | --- |",
     "example": 205
   },
+  {
```
Comment (UziTech) [inline test/specs/gfm/gfm.0.29.json:50]: You shouldn't change the `test/specs/gfm/ or `test/spec/commonmark/` json files as they are generated from the spec.

You can add these to the `test/specs/new/` folder. Just add a `.md` file with the markdown and a `.html` file with the same name with the expected html.

**Review by UziTech — CHANGES_REQUESTED** [review]

File: test/specs/gfm/gfm.0.29.json (hunk 1 above)
Comment (EmmetZC) [reply test/specs/gfm/gfm.0.29.json:50]: Oh, I see.

Comment (EmmetZC) [reply test/specs/gfm/gfm.0.29.json]: Changes to `test/specs/gfm` reverted. 
Test cases migrated to `test/specs/new/table_following_text.*`

**Review by UziTech — APPROVED** [review]
:100:

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**